src/main/java/com/marklogic/kafka/connect/MarkLogicConfig.java

```diff
@@ -30,16 +31,28 @@ public class MarkLogicConfig extends AbstractConfig {
     public static final String SSL_HOST_VERIFIER = "ml.connection.customSsl.hostNameVerifier";
     public static final String SSL_MUTUAL_AUTH = "ml.connection.customSsl.mutualAuth";
 
-    private static final CustomRecommenderAndValidator SecurityContextTypeRV = new CustomRecommenderAndValidator("DIGEST", "BASIC", "CERTIFICATE", "KERBEROS", "NONE");
+    private static final CustomRecommenderAndValidator CONNECTION_HOST_RV = new CustomRecommenderAndValidator();
+    private static final CustomRecommenderAndValidator CONNECTION_PORT_RV = new CustomRecommenderAndValidator();
+    private static final CustomRecommenderAndValidator CONNECTION_SECURITY_CONTEXT_TYPE_RV = new CustomRecommenderAndValidator("DIGEST", "BASIC", "CERTIFICATE", "KERBEROS", "NONE");
+    private static final CustomRecommenderAndValidator CONNECTION_TYPE_RV = new CustomRecommenderAndValidator("DIRECT", "GATEWAY", "");
+    private static final CustomRecommenderAndValidator SSL_HOST_VERIFIER_RV = new CustomRecommenderAndValidator("ANY", "COMMON", "STRICT");
+
+    protected static final CustomRecommenderAndValidator DMSDK_BATCH_SIZE_RV = new CustomRecommenderAndValidator();
+    protected static final CustomRecommenderAndValidator DMSDK_THREAD_COUNT_RV = new CustomRecommenderAndValidator();
+    protected static final CustomRecommenderAndValidator BULK_DS_BATCH_SIZE_RV = new CustomRecommenderAndValidator();
```

Comment:
Check out the Kafka built-in Validator, ConfigDef.Range. Look in MarkLogicSourceConfig for an example.